File: Pie02/Pie02IronPython/Rules/MethodDeclarationRules.py
from Irony.Parsing import *

class MethodDeclarationRules(object):
    """description of class"""

    def __init__(self, grammar):
        self.MethodMember = NonTerminal("method_member")
        self.MethodMember.Rule = grammar.StatementExpression

        self.MethodMemberList = NonTerminal("method_member_list")
        self.MethodMemberList.Rule = grammar.MakeStarRule(self.MethodMemberList, self.MethodMember)

        self.MethodBody = NonTerminal("method_body")
        self.MethodBody.Rule = grammar.Indent + self.MethodMemberList + grammar.Dedent
        self.MethodBody.Rule |= grammar.Empty

        self.MethodDeclaration = NonTerminal("method_declaration")
        self.MethodDeclaration.Rule = grammar.ModifierList + grammar.Void + grammar.Identifier + grammar.GenericListOpt + grammar.LParens + grammar.RParens + grammar.Colon + grammar.Eos + self.MethodBody

        

File: Pie02/Pie02CS/Rules/MethodDeclarationRules.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

using Irony.Parsing;

namespace Pie02CS.Rules
{
    public class MethodDeclarationRules
    {

        public NonTerminal MethodDeclaration;

        public NonTerminal ConstructorDeclaration;

        public MethodDeclarationRules(PieGrammar grammar)
        {


            MethodDeclaration = new NonTerminal("method_declaration");
            MethodDeclaration.Rule = grammar.ModifierList + grammar.MethodType + grammar.Identifier + grammar.LParens + grammar.ParameterList + grammar.RParens + ":" + grammar.Eos + grammar.MethodBody;

            ConstructorDeclaration = new NonTerminal("constructor_declaration");
            ConstructorDeclaration.Rule = grammar.ModifierList + grammar.ToTerm("new") + grammar.LParens + grammar.ParameterList + grammar.RParens + ":" + grammar.Eos + grammar.MethodBody;

        }
    }
}
